spec/controllers/reviews_controller_spec.rb

```diff
@@ -0,0 +1,74 @@
+require 'spec_helper'
+
+describe ReviewsController do
+
+  describe 'POST :create' do
+
+    let(:video) { video = Fabricate(:video) }
+
+    context 'user is authorized' do
+      let(:user) { user = Fabricate(:user) }
+      before do
+        self.controller.login_user user
+      end
+
+      context 'review data is valid' do
+        before do
+          post :create, video_id: video, review: Fabricate.attributes_for(:review)
+        end
+        it 'creates a new review' do
+          expect(Review.count).to eq 1
+        end
+        it 'creates a new review associated with the video' do
+          expect(Review.last.video).to eq video
+        end
+        it 'creates a new review associated with the signed in user' do
+          expect(Review.last.user).to eq user
+        end
+        it 'sets the notice' do
+          expect(flash[:notice]).to eq 'Review posted successfully.'
+        end
+        it 'redirects to the video page' do
+          expect(response).to redirect_to video_path(video)
+        end
+      end
+
+      context 'review data is not valid' do
+        before do
+          post :create, video_id: video, review: { rating: 5 }
+        end
+        it 'does not create a new review' do
+          expect(Review.count).to eq 0
+        end
+        it 'sets the error message' do
+          expect(flash[:error]).to eq 'Review not saved.'
```

Comment:
Testing on flash messages in controller is cumbersome, because these most likely will change. Also I like testing them in integration tests rather - but that will be in next lesson.

So if you like it, I don't mind. Just it can set your test suite to be less flexible. If so I would test it on a presence of the flash[:error].